WooCommerce/Classes/ViewModels/Order Details/CardPaymentReadinessUseCase.swift

```diff
@@ -0,0 +1,67 @@
+import Foundation
+import Combine
+import Yosemite
+
+final class CardPaymentReadinessUseCase {
+    /// Represents the system's readiness to accept a card payment
+    ///
+    enum CardPaymentReadiness {
+        /// Current state is being fetched
+        case loading
+
+        /// Onboarding has been successfully completed, and reader connection can be triggered.
+        /// N.B. A reader may already be connected.
+        case ready
+
+        /// Onboarding is in an incomplete/error state, so should be displayed instead of attempting a reader connection
+        case onboardingRequired
+    }
+
+    @Published var readiness: CardPaymentReadiness = .loading
+
+    /// Observes the store's current CPP state.
+    ///
+    private let onboardingUseCase: CardPresentPaymentsOnboardingUseCase
+
+    private let stores: StoresManager
+
+    init(onboardingUseCase: CardPresentPaymentsOnboardingUseCase,
+         stores: StoresManager = ServiceLocator.stores) {
+        self.onboardingUseCase = onboardingUseCase
+        self.stores = stores
+    }
+
+    /// Checks whether there is a reader connected (implying that we're ready to accept payments)
+    /// If there's not, checks whether `CardPresentOnboardingState` is `.completed`
+    ///
+    func checkCardPaymentReadiness() {
+        onboardingUseCase.refresh()
+        let readerConnected = CardPresentPaymentAction.checkCardReaderConnected { connectPublisher in
+            // TODO: Use readerConnectedReadiness to preempt the onboarding readiness check.
+            // This requires a refactor of `checkCardReaderConnected` to emit an event for a connected reader.
```

Comment:
Would it make sense to log a ticket for the refactor suggested and link to it directly?

Reply:
Added:
- #6766